Src/DependencyCollector/Microsoft.AI.DependencyCollector/DependencyCollectorDiagnosticListener.cs

```diff
@@ -0,0 +1,189 @@
+        {
+            if (configuration == null)
+            {
+                configuration = TelemetryConfiguration.Active;
+            }
+
+            return DiagnosticListener.AllListeners.Subscribe(new DependencyCollectorDiagnosticListener(configuration));
+        }
+
+        /// <summary>
+        /// Source instrumentation header that is added by an application while making http requests and retrieved by the other application when processing incoming requests.
+        /// </summary>
+        internal const string SourceInstrumentationKeyHeader = "x-ms-request-source-ikey";
+
+        /// <summary>
+        /// Target instrumentation header that is added to the response and retrieved by the calling application when processing incoming responses.
+        /// </summary>
+        internal const string TargetInstrumentationKeyHeader = "x-ms-request-target-ikey";
+
+        /// <summary>
+        /// Standard parent Id header.
+        /// </summary>
+        internal const string StandardParentIdHeader = "x-ms-request-id";
+
+        /// <summary>
+        /// Standard root id header.
+        /// </summary>
+        internal const string StandardRootIdHeader = "x-ms-request-root-id";
+
+        private readonly ApplicationInsightsUrlFilter applicationInsightsUrlFilter;
+        private readonly TelemetryClient client;
+        private readonly ConcurrentDictionary<Guid, DependencyTelemetry> pendingTelemetry = new ConcurrentDictionary<Guid, DependencyTelemetry>();
+
+        internal DependencyCollectorDiagnosticListener(TelemetryConfiguration configuration)
+        {
+            this.client = new TelemetryClient(configuration);
+
+            // Since dependencySource is no longer set, sdk version is prepended with information which can identify whether RDD was collected by profiler/framework
+            // For directly using TrackDependency(), version will be simply what is set by core
+            string versionStr = typeof(DependencyCollectorDiagnosticListener).GetTypeInfo().Assembly.GetCustomAttributes()
+                    .OfType<AssemblyFileVersionAttribute>()
+                    .First()
+                    .Version;
+
+            Version version = new Version(versionStr);
+            this.client.Context.GetInternalContext().SdkVersion = "rddf" + version.ToString(3) + "-" + version.Revision;
+
+            this.applicationInsightsUrlFilter = new ApplicationInsightsUrlFilter(configuration);
+        }
+
+        public void OnNext(DiagnosticListener value)
+        {
+            // Comes from https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/DiagnosticsHandlerLoggingStrings.cs#L12
+            if (value.Name == "HttpHandlerDiagnosticListener")
+            {
+                value.SubscribeWithAdapter(this);
+            }
+        }
+
+        public void OnCompleted()
+        {
+        }
+
+        public void OnError(Exception error)
+        {
```

Comment:
EventSource is available in .NET Core, we use it extensively in each of the SDK's including the .NET Core SDK.
Each SDK has its own class for messages, the one in the dependency collector project is DependencyCollectorEventSource.